src/async_impl/client.rs

```diff
@@ -1424,6 +1438,11 @@ impl Future for PendingRequest {
                 self.client.accepts,
                 self.timeout.take(),
             );
+
+            if self.client.error_for_status {
+                res.error_for_status_ref()?;
+            }
+
```

Comment:
I used `error_for_status_ref` here, because it is cleaner than

```rust
let res = {
    if self.client.error_for_status {
        res.error_for_status()?
    } else {
        res
    }
};
```